### PR TITLE
Fix poetry-core migration failure for projects not using major.minor.patch version references

### PR DESCRIPTION
**Pull request by srmedava**
1) Fixed code to tackle the failure when the patch version is not present in the toml file
2) Added 2 BDD tests for the cases identified in this issue

### CODE REVIEW

File: habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/migration/CustomPoetrycoreVersionMigration.java [hunk 1]
```diff
@@ -159,4 +176,68 @@ private boolean isPoetrycoreUpgradeRequired(Map.Entry<String, Object> dependency
         }
         return false;
     }
+
```
Comment (d-ryan-ashcraft) [inline habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/migration/CustomPoetrycoreVersionMigration.java:132]: S: In other places, we use semver4j and it handles a lot of this complexity and probably will cover a lot of edges too: https://github.com/vdurmont/semver4j

**Review by d-ryan-ashcraft — CHANGES_REQUESTED** [review]

**Comment by cpointe-ibllanos** [comment]
Looks good, but I'll let Ryan give out the approval. I can vouch for semver4j -- helped me cut out a few hundred lines of reinventing the wheel.

File: habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/migration/CustomPoetrycoreVersionMigration.java (hunk 1 above)
Comment (srmedava) [reply habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/migration/CustomPoetrycoreVersionMigration.java:132]: Used semver4j and cleaned up a lot of the code.

File: habushu-maven-plugin/src/test/resources/specifications/poetry-core-version-migration.feature [hunk 2]
```diff
@@ -2,22 +2,32 @@ Feature: Poetry-core version migration
...
+    Then no update was performed for poetry-core version
 
-Scenario: No migration will be performed when the poetry-core version of 2.0.0
-Given an existing pyproject.toml file with poetry-core version of 2.0.0 in the build-system group
-When Habushu poetry core migration executes
-Then no update was performed for poetry-core version
\ No newline at end of file
+  Scenario: No migration will be performed when the poetry-core version of 2.0.0
+    Given an existing pyproject.toml file with poetry-core version of 2.0.0 in the build-system group
+    When Habushu poetry core migration executes
```
Comment (d-ryan-ashcraft) [inline habushu-maven-plugin/src/test/resources/specifications/poetry-core-version-migration.feature:22]: I: recommend ignoring whitespace

**Review by d-ryan-ashcraft — APPROVED** [review]

File: habushu-maven-plugin/src/test/resources/specifications/poetry-core-version-migration.feature (hunk 2 above)
Comment (srmedava) [reply habushu-maven-plugin/src/test/resources/specifications/poetry-core-version-migration.feature:22]: I don't understand this comment. Could you elaborate a little further? Will try to see if I can fix it.

Comment (srmedava) [reply habushu-maven-plugin/src/test/resources/specifications/poetry-core-version-migration.feature:22]: Not affecting any functionality, so merging it. will fix in other tickets if this becomes an issue.

Comment (d-ryan-ashcraft) [reply habushu-maven-plugin/src/test/resources/specifications/poetry-core-version-migration.feature:22]: I - Informational.  This is for someone else reading this.  80% of the changes in the file where whitespace, so I'm just trying to help the next person out interpreting it.

Comment (srmedava) [reply habushu-maven-plugin/src/test/resources/specifications/poetry-core-version-migration.feature:22]: Got it. Thanks.

**Pull request merged**